data_structures/src/chain.rs

```diff
@@ -179,10 +179,9 @@ pub struct BlockInChain {
     pub data_request_pool: DataRequestPool,
 }
 
-/// Any reference to a Hashable type is also Hashable
-impl<'a, T: Hashable> Hashable for &'a T {
+impl<T: AsRef<[u8]>> Hashable for T {
     fn hash(&self) -> Hash {
-        (*self).hash()
+        calculate_sha256(self.as_ref()).into()
```

Comment:
Could this hinder any attempt to implement #510? That's super important!

Reply:
hmm I don't think so :thinking: `Transaction` does not (nor can) implement `AsRef<[u8]>` so it will always use its own `Hashable` impl. That's what you mean?